src/predictor/gpu_predictor.cu

```diff
@@ -324,21 +336,27 @@ class GPUPredictor : public xgboost::Predictor {
                 h_nodes.begin() + h_tree_segments[tree_idx - tree_begin]);
     }
 
-    size_t i_batch = 0;
     size_t batch_offset = 0;
     for (auto &batch : dmat->GetRowBatches()) {
-      CHECK(i_batch == 0 || devices_.Size() == 1) << "External memory not supported for multi-GPU";
-      // out_preds have been sharded and resized in InitOutPredictions()
-      batch.offset.Shard(GPUDistribution::Overlap(devices_, 1));
+      GPUSet devices = GPUSet::All(param_.gpu_id, param_.n_gpus, batch.Size());
+      ConfigureShards(devices);
+
+      batch.offset.Reshard(GPUDistribution::Overlap(devices_, 1));
+
       std::vector<size_t> device_offsets;
       DeviceOffsets(batch.offset, &device_offsets);
       batch.data.Reshard(GPUDistribution::Explicit(devices_, device_offsets));
```

Comment:
Is the granularity handled correctly?  Could you make a test for predicting multiclass output showing the correctness for granularity? Is the resharding in `InitOutPredictions()` still needed?

Reply:
Should be as `n_classes` is taken into consideration for the explicit offsets. I'll look at adding a test for multiclass.

Removed the sharding in `InitOutPredictions()`.

Reply:
Changed the test a bit to test multi-class output.